glean_parser/lint.py

```diff
@@ -130,16 +150,41 @@ def check_bug_number(metric):
         )
 
 
+def check_valid_in_baseline(metric, parser_config={}):
+    allow_reserved = parser_config.get("allow_reserved", False)
+
+    if not allow_reserved and "baseline" in metric.send_in_pings:
+        yield (
+            f"The baseline ping is Glean-internal. "
+            "User metrics should go into the 'metrics' ping or custom pings."
+        )
+
+
+def check_misspelled_pings(metric, parser_config={}):
+    builtin_pings = ["metrics", "events"]
+
+    for ping in metric.send_in_pings:
+        for builtin in builtin_pings:
+            distance = _hamming_distance(ping, builtin)
+            if distance == 1:
+                yield (f"Ping '{ping}' seems misspelled. Did you mean '{builtin}'?")
+
+
```

Comment:
I had thought we could do this for all of the "enumeration-like" parameters: `type`, `lifetime`, `time_unit`, `memory_unit`, and `histogram_type`.  Up to you whether you want to do that now or in a follow-on bug.

Reply:
hm, guess that would make sense. I can take it here.

Reply:
Actually, my suggestion doesn't make much sense.  These others are exactly required already.  :)

Reply:
Where it might get interesting is for things where we allow people to write their own strings. Like `custom_distribution`'s "unit" or any `category` name

Reply:
Spell checker?  (j/k)

Reply:
Sounds like something for a followup